### ISSUE TITLE
SSH Handshake failure since 1.20

### ISSUE THREAD

**Issue by geNAZt** [istio/istio#49826]
### Is this the right place to submit this?

- [X] This is not a security vulnerability or a crashing bug
- [X] This is not a question about how to use Istio

### Bug Description

When upgrading form 1.19.6 to 1.20.3 (we also tried earlier 1.20 versions) we see that our flux setup (source-controller) sees SSH handshake errors like this:

'failed to checkout and determine revision: unable to clone ''ssh://git@github.com/paymenttools/deployment-pci.git'':
      ssh: handshake failed: read tcp 10.70.34.10:57140->140.82.121.3:22: read: connection
      reset by peer'
      
This only appeared after a full update to the new sidecar version.

### Version

```Text
istioctl version:

client version: 1.20.3
control plane version: 1.20.3
data plane version: 1.19.6 (74 proxies), 1.20.3 (39 proxies)

k8s version:

Client Version: v1.29.2
Kustomize Version: v5.0.4-0.20230601165947-6ce0bf390ce3
Server Version: v1.29.2-gke.1060000
```

### Additional Information

[bug-report.zip](https://github.com/istio/istio/files/14549490/bug-report.zip)

**Comment by geNAZt** [comment]
We have defined this ServiceEntry:

```yaml
apiVersion: networking.istio.io/v1beta1
kind: ServiceEntry
metadata:
  annotations:
    meta.helm.sh/release-name: source-controller-networking
    meta.helm.sh/release-namespace: flux-system
  creationTimestamp: "2023-07-04T06:45:07Z"
  generation: 1
  labels:
    app: source-controller
    app.kubernetes.io/instance: source-controller-networking
    app.kubernetes.io/managed-by: Helm
    app.kubernetes.io/name: source-controller
    helm.sh/chart: network-6.1.0
    helm.toolkit.fluxcd.io/name: source-controller-networking
    helm.toolkit.fluxcd.io/namespace: flux-system
    owner: dev-team-fire-brigade
  name: source-controller-networking-github.com-22
  namespace: flux-system
  resourceVersion: "709994182"
  uid: d10bc20a-d649-4337-9d2b-919c6ba00359
spec:
  exportTo:
  - .
  hosts:
  - github.com
  location: MESH_EXTERNAL
  ports:
  - name: tcp
    number: 22
    protocol: TCP
  resolution: DNS
```

and for port 443:

```yaml
apiVersion: networking.istio.io/v1beta1
kind: ServiceEntry
metadata:
  annotations:
    meta.helm.sh/release-name: source-controller-networking
    meta.helm.sh/release-namespace: flux-system
  creationTimestamp: "2023-07-04T06:45:07Z"
  generation: 1
  labels:
    app: source-controller
    app.kubernetes.io/instance: source-controller-networking
    app.kubernetes.io/managed-by: Helm
    app.kubernetes.io/name: source-controller
    helm.sh/chart: network-6.1.0
    helm.toolkit.fluxcd.io/name: source-controller-networking
    helm.toolkit.fluxcd.io/namespace: flux-system
    owner: dev-team-fire-brigade
  name: source-controller-networking-github.com-443
  namespace: flux-system
  resourceVersion: "709994185"
  uid: ee4805c1-5526-409c-aa33-57d7a4b17ac4
spec:
  exportTo:
  - .
  hosts:
  - github.com
  location: MESH_EXTERNAL
  ports:
  - name: https
    number: 443
    protocol: TLS
  resolution: DNS
```

**Comment by hzxuzhonghu** [comment]
I do think this is mostly service merge bug, you can workaround by append 443 to 80

The fix is https://github.com/istio/istio/pull/48896

**Comment by geNAZt** [comment]
Thanks for the workaround, is there already a release planned with which this fix will land?

I guess we can still stay on 1.19 until this is fixed since 1.19 still gets security updates.

**Comment by hzxuzhonghu** [comment]
Not backported yet
